### PR TITLE
Add account existency check at salesforce before it's creation to avoid duplicity

### PR DESCRIPTION
**Pull request by beligante**
Since we have some accounts already recorded at salesforce, we need to check if for a given telephone number there are an respective account with it at salesforce. With this check we avoid duplicated accounts there

### CODE REVIEW

File: apps/re/test/partner_brokers/job_queue_test.exs [hunk 1]
```diff
@@ -11,15 +12,75 @@ defmodule Re.PartnerBrokers.JobQueueTest do
...
+    @uri_query %URI{path: "/api/v1/query"}
+
+    test "should create account when it doesn't exists at salesforce" do
+      mock(HTTPoison, [post: 3], fn
+        @uri_query, _, _ ->
+          @empty_response
+
+        @uri_account, _, _ ->
+          {:ok, %{status_code: 200, body: ~s({"success": true, "id": "123", "errors": []})}}
+      end)
```
Comment (rhnonose) [inline apps/re/test/partner_brokers/job_queue_test.exs]: You're adding a mock with 2 calls but asserting only one.
Is both being called? Then, both should be asserted.

File: apps/re/test/partner_brokers/job_queue_test.exs [hunk 2]
```diff
@@ -54,14 +115,122 @@ defmodule Re.PartnerBrokers.JobQueueTest do
...
         ^uri,
         ^body,
         [{"Authorization", ""}, {"Content-Type", "application/json"}]
       ])
     end
+
+    test "should not create account when it already exists at salesforce" do
+      mock(HTTPoison, [post: 3], fn
+        @uri_query, _, _ -> @single_response
+      end)
```
Comment (rhnonose) [inline apps/re/test/partner_brokers/job_queue_test.exs]: This syntax is for declaring multiple mocks for the same module/function.
You could write a shorter version like this:
`mock(HTTPoison, :post, @single_response)`
(double-check with other examples).

File: apps/re/test/partner_brokers/job_queue_test.exs [hunk 3]
```diff
@@ -54,14 +115,122 @@ defmodule Re.PartnerBrokers.JobQueueTest do
...
+        ^body,
+        [{"Authorization", ""}, {"Content-Type", "application/json"}]
+      ])
+    end
+
+    @tag capture_log: true
+    test "should consider the first salesforceid when there are multiple results for the same account phone" do
+      mock(HTTPoison, [post: 3], fn
+        @uri_query, _, _ -> @multiple_response
+      end)
```
Comment (rhnonose) [inline apps/re/test/partner_brokers/job_queue_test.exs]: Same as previous test.

File: apps/re/test/partner_brokers/job_queue_test.exs [hunk 4]
```diff
@@ -54,14 +115,122 @@ defmodule Re.PartnerBrokers.JobQueueTest do
...
+        ^body,
+        [{"Authorization", ""}, {"Content-Type", "application/json"}]
+      ])
+    end
+
+    @tag capture_log: true
+    test "should log that an specific id at salesforce have duplicated phone number" do
+      mock(HTTPoison, [post: 3], fn
+        @uri_query, _, _ -> @multiple_response
+      end)
```
Comment (rhnonose) [inline apps/re/test/partner_brokers/job_queue_test.exs]: Same as previous test.

File: apps/re/lib/partner_brokers/partner_brokers.ex [hunk 5]
```diff
@@ -42,7 +42,21 @@ defmodule Re.PartnerBrokers do
 
   def create_account(uuid) do
     with {:ok, user} <- Users.get_preloaded(uuid, :districts),
-         {:ok, %{"id" => id, "success" => true}} <- Salesforce.create_account(user),
+         do: search_account_at_salesforce(user)
+  end
+
+  defp search_account_at_salesforce(%{phone: phone} = user) do
```
Comment (pmargreff) [inline apps/re/lib/partner_brokers/partner_brokers.ex]: `search_account_by_phone` is an option

**Pull request merged**